DigiIoT.Maui/Platforms/iOS/Bluetooth/BeaconAdvertiseService.cs

```diff
@@ -0,0 +1,139 @@
+			//	CBAdvertisement.DataLocalNameKey, identifier,
+			//	CBAdvertisement.DataManufacturerDataKey, nsData
+			//);
+			//
+			// However, the 'DataManufacturerDataKey' tag is not allowed for advertisement, so
+			// custom beacons are not available in iOS.
+
+			throw new NotSupportedException("Manufacturer data AD is not supported in iOS.");
+		}
+
+		/// <summary>
+		/// Starts advertising an iBeacon with the provided parameters.
+		/// </summary>
+		/// <param name="identifier">Identifier or local name of the beacon.</param>
+		/// <param name="uuid">UUID of the iBeacon.</param>
+		/// <param name="major">Major value of the iBeacon.</param>
+		/// <param name="minor">Minor value of the iBeacon.</param>
+		public partial void StartAdvertisingIBeacon(string identifier, string uuid, ushort major, ushort minor)
+		{
+			// Initialize vars.
+			var proximityUUID = new NSUuid(uuid);
+
+			if (UIDevice.CurrentDevice.CheckSystemVersion(17, 0))
+			{
+				// For iOS 17.0 and later
+				var beaconData = new NSMutableDictionary();
+
+				var serviceUUIDs = NSArray.FromObjects(new CBUUID[] { CBUUID.FromString(uuid) });
+
+				beaconData.Add(CBAdvertisement.DataServiceUUIDsKey, serviceUUIDs);
+				beaconData.Add(new NSString(PARAM_MAJOR), NSNumber.FromUInt16(major));
+				beaconData.Add(new NSString(PARAM_MINOR), NSNumber.FromUInt16(minor));
+				beaconData.Add(CBAdvertisement.DataLocalNameKey, new NSString(identifier));
+
+				_advertisementData = NSDictionary.FromDictionary(beaconData);
+			}
+			else if (UIDevice.CurrentDevice.CheckSystemVersion(13, 0))
+			{
+				// For iOS 13.0 to 16.x
+				var beaconIdentityConstraint = new CLBeaconIdentityConstraint(proximityUUID, major, minor);
+				var beaconRegion = new CLBeaconRegion(beaconIdentityConstraint, identifier);
+				_advertisementData = beaconRegion.GetPeripheralData(null);
+			}
+			else
+			{
+				// For iOS 12.x and earlier
+				var beaconRegion = new CLBeaconRegion(proximityUUID, major, minor, identifier);
+				_advertisementData = beaconRegion.GetPeripheralData(null);
+			}
+
+			// Start advertising data.
+			StartAdvertisingInternal();
+		}
+
+		/// <summary>
+		/// Starts advertising data.
+		/// </summary>
+		private void StartAdvertisingInternal()
+		{
+			// Initialize peripheral manager if not already done.
+			if (_peripheralManager == null)
+			{
+				_peripheralManager = new CBPeripheralManager();
+				_peripheralManager.StateUpdated += (sender, e) =>
+				{
+					if (_peripheralManager.State == CBManagerState.PoweredOn && _advertisementData != null)
+					{
+						_peripheralManager.StartAdvertising(_advertisementData);
+					}
+				};
+			}
+
+			if (_peripheralManager.State == CBManagerState.PoweredOn && _advertisementData != null)
+			{
+				_peripheralManager.StartAdvertising(_advertisementData);
+			}
+		}
+
+		/// <summary>
+		/// Stops advertising.
+		/// </summary>
+		public partial void StopAdvertising()
+		{
+			_peripheralManager?.StopAdvertising();
```

Comment:
I would do a couple of things here. First is to set '_advertisementData' to null so that '_peripheralManager.StateUpdated' callback won't trigger an 'StartAdvertising' action when state is 'PoweredOn' again. And, if possible, remove the callback and redo a bit the previous code:

			// Initialize peripheral manager if not already done.
			if (_peripheralManager == null)
			{
				_peripheralManager = new CBPeripheralManager();
			}

			_peripheralManager.StateUpdated += (sender, e) =>
			{
				if (_peripheralManager.State == CBManagerState.PoweredOn && _advertisementData != null)
				{
					_peripheralManager.StartAdvertising(_advertisementData);
				}
			};
			
			if (_peripheralManager.State == CBManagerState.PoweredOn && _advertisementData != null)
			{
				_peripheralManager.StartAdvertising(_advertisementData);
			}